elizaOS/eliza
# add an client-direct endpoint to get memories by agentid and roomid

# Relates to:



# Risks

Low. This change adds a new API endpoint for fetching memories, which should not affect existing functionality.
# Background

## What does this PR do?
This PR adds a new API endpoint /agents/:agentId/:roomId/memories to fetch memories for a specific agent and room. This enhancement supports improved web-based chat capabilities with Eliza characters.

## What kind of change is this?
Features (non-breaking change which adds functionality)

## Why are we doing this? Any context or related work?
Mee.fun a hosted Eliza service, want to add a web-based chat UI.  In order to do this history of the chat needs to be retrievable.

# Documentation changes needed?
My changes require a change to the project documentation. The new API endpoint should be documented in the appropriate API documentation.

# Testing
## Where should a reviewer start?

1. Review the new route added in packages/client-direct/src/api.ts
2. Check the structure of the response in the /agents/:agentId/:roomId/memories endpoint

## Detailed testing steps

1. Start the server
2. Use a tool like Postman or curl to make a GET request to /agents/{agentId}/{roomId}/memories, replacing {agentId} and {roomId} with valid IDs
3. Verify that the response includes the correct structure with agentId, roomId, and an array of memories
4. Check that each memory in the response includes all the fields from the Memory interface
5. Test with invalid agentId to ensure a 404 response is returned
6. Test with a valid agentId but no memories to ensure an empty array is returned

## Discord username
treppers
